hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -502,12 +521,16 @@ private void writeToBuffer(HoodieRecord<T> record) {
       record.seal();
     }
     // fetch the ordering val first in case the record was deflated.
-    final Comparable<?> orderingVal = record.getData().getOrderingValue();
-    Option<IndexedRecord> indexedRecord = getIndexedRecord(record);
+    final Comparable<?> orderingVal = ((HoodieRecordPayload)record.getData()).getOrderingValue();
+    Option<HoodieRecord> indexedRecord = prepareRecord(record);
     if (indexedRecord.isPresent()) {
       // Skip the ignored record.
-      if (!indexedRecord.get().equals(IGNORE_RECORD)) {
-        recordList.add(indexedRecord.get());
+      try {
+        if (indexedRecord.isPresent() && !indexedRecord.get().isIgnoredRecord(tableSchema, config.getProps())) {
+          recordList.add(indexedRecord.get());
+        }
+      } catch (IOException e) {
+        LOG.error("Error writing record  " + indexedRecord.get(), e);
```

Comment:
why this could throw IOException? thought it's just checking some configs.

Reply:
isIgnoredRecord will deserializes(throw IOException) data to get indexRecord and check if it is EmptyRecord